nibabel/nifti1.py

```diff
@@ -1260,7 +1260,7 @@ def set_dim_info(self, freq=None, phase=None, slice=None):
         This is stored in one byte in the header
         '''
         for inp in (freq, phase, slice):
-            if inp not in (None, 0, 1, 2):
+            if inp is not None and inp not in (0, 1, 2):
```

Comment:
I don't get that warning myself, but that's fine - maybe add a comment above to explain.